include/deal.II/base/function.templates.h

```diff
@@ -26,11 +26,6 @@
 
 DEAL_II_NAMESPACE_OPEN
 
-
-template <int dim, typename Number>
-const unsigned int Function<dim, Number>::dimension;
```

Comment:
You are correct: we cannot use it as an lvalue unless we do this. Today I learned something new about this corner of C++ :)

I suppose I am used to (being new to C++) declaring these things as `constexpr` and ignoring such details.

For posterity: it is always good to refer to the master:

http://www.stroustrup.com/bs_faq2.html#in-class